app/vmctl/native/client.go

```diff
@@ -89,6 +91,12 @@ func (c *Client) ImportPipe(ctx context.Context, dstURL string, pr *io.PipeReade
 	if err != nil {
 		return fmt.Errorf("cannot create import request to %q: %s", c.Addr, err)
 	}
+
+	err = parseAndSetHeaders(c, req)
```

Comment:
It is not a good idea to use the client as a function param, you can just parse headers and set them into a request right after request initialization